### PR TITLE
Allow for non unique state IDs

### PR DESCRIPTION
**Pull request by inxilpro**
Right now, Verbs snapshots have a unique `id` primary key, which means that you can only have one of any given ID snapshotted at any given time. This is fine if you use snowflakes for everything, but if you don't that can be an issue.

@skylerkatz and I paired on this for a while, and I know there were issues remaining, but I can't remember what they are. I'm going to post this up as a draft while I review it and maybe add some more tests.

### CODE REVIEW

**Review by skylerkatz — APPROVED** [review]
Overall this makes sense to me, and it does resolve our issue with using _legacy_ model records as state ids.

I still don't love the explicit reliance on `Snowflakes` even if you have configured some other key generation. But, that is a discussion for a different day.

**Comment by inxilpro** [comment]
@skylerkatz I forgot to push up a couple changes. Can you take another look?

Also—I agree that we should think about allowing Verbs to run without snowflakes at all. Let's discuss at some point.

**Review by DanielCoulbourne — APPROVED** [review]
LGTM

File: src/Lifecycle/MetadataManager.php [hunk 1]
```diff
@@ -42,15 +44,22 @@ public function getLastResults(Event $event): Collection
...
-    public function getEphemeral(Event $event, ?string $key = null, mixed $default = null): mixed
+    public function getEphemeral(Event|State $target, ?string $key = null, mixed $default = null): mixed
     {
-        return data_get($this->ephemeral[$event] ?? [], $key, $default);
+        $ephemeral = data_get($this->ephemeral[$target] ?? [], $key, $not_found = new stdClass());
+
+        if ($ephemeral === $not_found) {
+            $this->setEphemeral($target, $key, $default);
+            $ephemeral = $default;
+        }
```
[the remark this answers is not in the document]
Comment (skylerkatz) [reply src/Lifecycle/MetadataManager.php:54]: I think it is fine

**Pull request merged**